### PR TITLE
Implemented validation on start date

### PR DESCRIPTION
**Pull request by SamGodwin2**
### What is the context of this PR?
Adds the start date field to the date validation pills and allows the author to publish a schema with a date range validation.

### How to review 
Tests pass, front-end looks good and the e2e implementation works correctly.

### CODE REVIEW

**Comment by SamGodwin2** [comment]
[Validation against completion date - latest date (4)](https://trello.com/c/ugmw4Sb9/588-validation-against-completion-date-latest-date-4)

**Comment by SamGodwin2** [comment]
[Validation - against completion date for earliest date (2)](https://trello.com/c/p0LDbXMP/705-validation-against-completion-date-for-earliest-date-2)

File: cypress/integration/authenticated/validation_spec.js [hunk 1]
```diff
@@ -372,6 +373,7 @@ describe("Answer Validation", () => {
 
       it("should update the custom value", () => {
         toggleCheckboxOn("@latestDateToggle");
+        switchPilltab("Custom");
```
Comment (tgandrews) [inline cypress/integration/authenticated/validation_spec.js]: Do we need a test for "Now"?

File: src/components/Validation/Date/DateValidation.js [hunk 2]
```diff
@@ -193,13 +202,14 @@ class DateValidation extends React.Component {
             </RelativePositionSelect>
             <PathEnd />
           </AlignedColumn>
-          <Column cols={8}>
+          <Column cols={9}>
```
Comment (adammahmood) [inline src/components/Validation/Date/DateValidation.js]: This has implications on the size of the content picker select box.  We should ensure the size of the content picker select box in the Metadata and Previous Answer tabs remains the same as pre-prod.

File: src/components/Validation/ValidationPills.js [hunk 3]
```diff
@@ -22,12 +22,24 @@ export const ValidationPills = ({
...
   PreviousAnswer,
   Metadata,
+  Now,
   Custom
 }) => (
   <Pills
     value={entityType}
     onChange={onEntityTypeChange}
     options={compact([
+      isFunction(Now)
```
Comment (adammahmood) [inline src/components/Validation/ValidationPills.js]: A unit test for this should be added to ValidationPills.test.js

File: cypress/integration/authenticated/validation_spec.js (hunk 1 above)
Comment (adammahmood) [reply cypress/integration/authenticated/validation_spec.js]: Yes. I agree. We should add tests for "Now"

File: src/components/Validation/Date/DatePreview.js [hunk 4]
```diff
@@ -35,7 +35,8 @@ const DatePreview = ({
     [entityTypes.PREVIOUS_ANSWER]: previousAnswer
       ? previousAnswer.displayName
       : "",
-    [entityTypes.METADATA]: metadata ? metadata.displayName : ""
+    [entityTypes.METADATA]: metadata ? metadata.displayName : "",
+    [entityTypes.NOW]: "(Start date)"
```
Comment (adammahmood) [inline src/components/Validation/Date/DatePreview.js]: We should add unit tests for this.

File: src/components/Validation/Date/DateValidation.js (hunk 2 above)
Comment (SamGodwin2) [reply src/components/Validation/Date/DateValidation.js]: It does change the length of the content picker box slightly but it was the easiest way to make the pillbox long enough to fit the new options. Talked it over with Hamish and this seemed the simplest solution.

**Pull request merged**